orchestrator/graphql/schemas/subscription.py

```diff
@@ -123,10 +150,27 @@ async def depends_on_subscriptions(
         sort_by: list[GraphqlSort] | None = None,
         first: int = 10,
         after: int = 0,
+        depends_on_filter: SubscriptionRelationFilter | None = None,
     ) -> Connection[Annotated["SubscriptionInterface", strawberry.lazy(".subscription")]]:
         from orchestrator.graphql.resolvers.subscription import resolve_subscriptions
 
-        subscriptions = await info.context.core_depends_on_subs_loader.load((self.subscription_id, None))
+        _depends_on_filter = depends_on_filter if depends_on_filter else SubscriptionRelationFilter()
+
+        async def get_depends_on_subscriptions_from_loader(
+            subscription_ids: list[UUID],
+            filter_statuses: tuple[str, ...],
+        ) -> list[list[SubscriptionTable]]:
+            load_mapping = [(sub_id, filter_statuses) for sub_id in subscription_ids]
+            return await info.context.core_depends_on_subs_loader.load_many(load_mapping)
+
+        subscriptions = await get_recursive_relations(
+            [self.subscription_id],
+            tuple(_depends_on_filter.statuses or ()),
+            _depends_on_filter.recurse_product_types or [],
+            _depends_on_filter.recurse_depth_limit,
+            get_depends_on_subscriptions_from_loader,
+        )
+
```

Comment:
You can move this into a local helper function in this module, something like

```py
    async def _load_recursive_relations(subscription_id: UUID, relation_filter: SubscriptionRelationFilter | None, data_loader: Callable[...]) -> list[SubscriptionTable]:
        sub_relation_filter = relation_filter or SubscriptionRelationFilter()

        async def get_subscriptions_from_loader(subscription_ids: list[UUID], filter_statuses: tuple[str, ...]) -> list[list[SubscriptionTable]]:
            load_mapping = [(sub_id, filter_statuses) for sub_id in subscription_ids]
            return await data_loader.load_many(load_mapping)

        subscriptions = await get_recursive_relations(
            [subscription_id],
            tuple(sub_relation_filter.statuses or ()),
            sub_relation_filter.recurse_product_types or [],
            sub_relation_filter.recurse_depth_limit,
            get_subscriptions_from_loader,
        )
```


And then use it here like 

```suggestion
        subscriptions = await _load_recursive_relations(self.subscription_id, depends_on_filter, info.context.core_depends_on_subs_loader)

```